### PR TITLE
Upgrade to Kotlin 2.0.0

### PR DESCRIPTION
**Pull request by DRSchlaubi**
- Update Gradle to 8.8
- Update kotlinx.coroutines to 1.8.1
- Update stately to 2.0.7
- Update kotlinx-atomicfu to 0.24.0
- Update kotlinx-serialization to 1.7.0
- Update kotlinx-benchmark to 0.4.11
- Update kotlin-logging to 6.0.9
- Update slf4j-api to 2.0.13
- Update lettuce to 6.3.2.RELEASE
- Update binary-compatibility-validator to 0.15.0-Beta.2
- Update dokka 1.9.20
- Update gradle-maven-publish-plugin to 0.28.0
- Enable KLIB BCV

### CODE REVIEW

File: buildSrc/src/main/kotlin/multiplatform-module.gradle.kts [hunk 1]
```diff
@@ -64,8 +64,18 @@ tasks {
     getByName<KotlinJvmTest>("jvmTest") {
         useJUnitPlatform()
     }
+    withType<KotlinNativeSimulatorTest>() {
+        enabled = false
+    }
```
Comment (lukellmann) [inline buildSrc/src/main/kotlin/multiplatform-module.gradle.kts:69]: why this?

**Review by lukellmann — APPROVED** [review]

File: buildSrc/src/main/kotlin/multiplatform-module.gradle.kts (hunk 1 above)
Comment (DRSchlaubi) [reply buildSrc/src/main/kotlin/multiplatform-module.gradle.kts:69]: The cloud based MacOS runners don't have the simulator runtimes installed

Comment (lukellmann) [reply buildSrc/src/main/kotlin/multiplatform-module.gradle.kts:69]: i see

File: api/src/commonMain/kotlin/dev/kord/cache/api/ConcurrentHashMap.kt
Comment (lukellmann) [inline api/src/commonMain/kotlin/dev/kord/cache/api/ConcurrentHashMap.kt:1]: would it be feasible to do it like kord instead? https://github.com/kordlib/kord/blob/b8bde7adde1c59b41e65ec71b586715f893035bd/common/src/commonMain/kotlin/ConcurrentHashMap.kt

Comment (DRSchlaubi) [reply api/src/commonMain/kotlin/dev/kord/cache/api/ConcurrentHashMap.kt:1]: we already discussed this https://github.com/kordlib/cache/pull/12#discussion_r1470148261

Comment (lukellmann) [reply api/src/commonMain/kotlin/dev/kord/cache/api/ConcurrentHashMap.kt:1]: forgot, alright

**Pull request merged**